src/Agent.Sdk/SecretMasking/ILoggedSecretMasker.cs

```diff
@@ -19,5 +25,8 @@ public interface ILoggedSecretMasker : ISecretMasker, IDisposable
         void AddValue(String value, string origin);
         void AddValueEncoder(ValueEncoder encoder, string origin);
         void SetTrace(ITraceWriter trace);
+
+        void EnableTelemetry();
+        void PublishTelemetry(PublishSecretMaskerTelemetryAction publishAction);
```

Comment:
>PublishTelemetry [](http://example.com/codeflow?start=13&length=16)

It occurs to me you could parameterize this method to include MaxDetectionsPerTelemetryEvent and MaxTelemetryDetectionEvents as additional arguments. The method could return a boolean indicating whether the publish step was complete, i.e., all things were successfully published given the paramterized constraints.

You could consider providing optional values for the max limits.

I like this idea because it codifies a useful contract around use of publish telemetry that is otherwise only expressed in the OssSecretMasker instance.

Reply:
Done.